### PR TITLE
Tests for nullable reference types in Span apis

### LINKED ISSUE #34358: Add more comprehensive tests for nullable reference types in Span apis
Missed cases: https://github.com/dotnet/corefx/issues/33710
Fixes for coverage: https://github.com/dotnet/coreclr/pull/21232

### PR DESCRIPTION
**Pull request by Anipik**
Fixes #34358 #33710

### CODE REVIEW

File: src/System.Memory/tests/ReadOnlySpan/IndexOf.T.cs [hunk 1]
```diff
@@ -193,5 +193,13 @@ public static void IndexOfNull_String(string[] spanInput, int expected)
...
             Assert.Equal(expected, theStrings.IndexOf((string)null));
         }
+
+        [Theory]
+        [MemberData(nameof(TestHelpers.ContainsNullData), MemberType = typeof(TestHelpers))]
+        public static void ContainsNull_String(string[] spanInput, bool expected)
+        {
+            ReadOnlySpan<string> theStrings = spanInput;
+            Assert.Equal(expected, theStrings.Contains(null));
+        }
```
Comment (stephentoub) [inline src/System.Memory/tests/ReadOnlySpan/IndexOf.T.cs]: Where are the rest of the Contains tests?  This file only contains IndexOf tests.

File: src/System.Memory/tests/ReadOnlySpan/IndexOfAny.T.cs [hunk 2]
```diff
@@ -963,5 +963,21 @@ public static void MakeSureNoChecksGoOutOfRangeIndexOfAny_ManyString()
...
+            ReadOnlySpan<string> theStrings = spanInput;
+            Assert.Equal(expected, theStrings.IndexOfAny(searchInput));
+        }
+
+        [Theory]
+        [MemberData(nameof(TestHelpers.IndexOfAnyNullData), MemberType = typeof(TestHelpers))]
+        public static void IndexOfAnyNull_String(string[] spanInput, string[] searchInput, int expected)
+        {
+            ReadOnlySpan<string> theStrings = spanInput;
+            Assert.Equal(expected, theStrings.IndexOfAny(searchInput[0], searchInput[1]));
```
Comment (stephentoub) [inline src/System.Memory/tests/ReadOnlySpan/IndexOfAny.T.cs]: Can you also add:
```C#
Assert.Equal(expected, theStrings.IndexOfAny((ReadOnlySpan<string>)searchInput));
```
?

Same for the other occurrences, where we can test both the overloads that accept multiple individual elements as well as a span of elements.

File: src/System.Memory/tests/TestHelpers.cs [hunk 3]
```diff
@@ -462,5 +520,35 @@ public static Memory<T> DangerousCreateMemory<T>(object obj, int offset, int len
...
+        {
+            { new string[] { "1", null, "2" }, new string[] { "1", null }, 1},
+            { new string[] { "1", null, "2" }, new string[] { null, null }, 1},
+            { new string[] { "1", null, "2" }, new string[] { "1", "2" }, 2},
+            { new string[] { null, null, "2" }, new string[] { null, "1" }, 1},
+            { new string[] { "1", null, "2" }, new string[] { "3", "4" }, -1},
+            { new string[] { null, null, "2" }, new string[] { null, "1" }, 1},
+            { new string[] { "1", "3", "2" }, new string[] { null, "1" }, 0},
+            { null, new string[] { null, "1" }, -1},
+        };
```
Comment (stephentoub) [inline src/System.Memory/tests/TestHelpers.cs]: Why do we need both LastIndexOfAnyNullSequenceData and LastIndexOfAnyNullData ?  We can't just have one test that takes the input data and calls both overloads?  Also, there are overloads that take three individual elements; where are those being tested?

File: src/System.Memory/tests/ReadOnlySpan/IndexOf.T.cs (hunk 1 above)
Comment (Anipik) [reply src/System.Memory/tests/ReadOnlySpan/IndexOf.T.cs]: We don't have contain tests for Span<T>. I will add that. but we Contains.T file that contains test for ReadOnlySpan. I will move that test to that file.

**Comment by ahsonkhan** [comment]
@Anipik, do you have an update on this PR?

**Comment by Anipik** [comment]
i am working on this, but it will take some time to complete it.

**Comment by ahsonkhan** [comment]
Friendly reminder, @Anipik :)

File: src/System.Memory/tests/TestHelpers.cs [hunk 4]
```diff
@@ -436,6 +464,33 @@ public static Memory<T> DangerousCreateMemory<T>(object obj, int offset, int len
...
+            { new string[] { "1", "3", "2" }, new string[] { null, null }, -1},
+
+            { new string[] { "1", "3", "2" }, new string[] { null, "1" }, 0},
+
+            { null, new string[] { "1", null, "2" }, -1},
+
+            { new string[] { "1", null, null }, new string[] { null, null, "2" }, 1},
+            { new string[] { null, null, null }, new string[] { null, null }, 0},
+
+            { new string[] { "1", "3", "2" }, null, 0},
```
Comment (ahsonkhan) [inline src/System.Memory/tests/TestHelpers.cs]: https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F38625~2Fmerge/test~2Ffunctional~2Fcli~2Finnerloop~2F/20190625.69/workItem/System.Memory.Tests

```text
IndexOfAnyNullSequence_String(spanInput: [\"1\", \"3\", \"2\"], searchInput: null, expected: 0)
```

```text
Unhandled Exception of Type Xunit.Sdk.EqualException
Message :
Assert.Equal() Failure
Expected: 0
Actual:   -1
Stack Trace :
   at System.SpanTests.ReadOnlySpanTests.IndexOfAnyNullSequence_String(String[] spanInput, String[] searchInput, Int32 expected) in /_/src/System.Memory/tests/ReadOnlySpan/IndexOfAny.T.cs:line 972

```

Are we sure we should expect 0?

cc @benaadams

Comment (benaadams) [reply src/System.Memory/tests/TestHelpers.cs]: Did previously; shouldn't now https://github.com/dotnet/corefx/issues/32334

i.e. it should be -1

**Pull request merged**